### PR TITLE
[ADD] Rate limiter

### PR DESCRIPTION
**Pull request by Plaenkler**

### CODE REVIEW

File: pkg/router/routes/api/job.go [hunk 1]
```diff
@@ -53,11 +59,16 @@ func CreateJob(w http.ResponseWriter, r *http.Request) {
...
 func UpdateJob(w http.ResponseWriter, r *http.Request) {
+	err := limit.IsOverLimit(r)
+	if err != nil {
+		w.WriteHeader(http.StatusTooManyRequests)
+		return
+	}
 	if r.Method != http.MethodPost {
 		w.WriteHeader(http.StatusMethodNotAllowed)
 		return
 	}
```
Comment (Valentin-Kaiser) [inline pkg/router/routes/api/job.go:70]: You can merge the two if statements and save the duplicate code

File: pkg/router/routes/api/config.go [hunk 2]
```diff
@@ -6,14 +6,20 @@ import (
 	"strconv"
 
 	"github.com/plaenkler/ddns/pkg/config"
+	"github.com/plaenkler/ddns/pkg/util/limit"
```
Comment (Valentin-Kaiser) [inline pkg/router/routes/api/config.go:9]: Why do you have a package path element called `util`? 
If I see it correctly, there are several packages in the folder. Shouldn't it be `utils` then, the plural?
I also think that this folder is superfluous and unnecessarily obscures the packages.

**Review by Valentin-Kaiser — CHANGES_REQUESTED** [review]

File: pkg/router/routes/api/job.go (hunk 1 above)
Comment (Plaenkler) [reply pkg/router/routes/api/job.go:70]: Those are different codes

File: pkg/router/routes/api/config.go (hunk 2 above)
Comment (Plaenkler) [reply pkg/router/routes/api/config.go]: Solved in [98dda37](https://github.com/Plaenkler/DDNS/pull/70/commits/98dda3783dfa196bfc4d3ed7d297b81c3336795e)

**Review by Valentin-Kaiser — APPROVED** [review]

File: pkg/ddns/providers/shared/request.go [hunk 3]
```diff
@@ -1,4 +1,4 @@
-package util
+package shared
```
Comment (Valentin-Kaiser) [inline pkg/ddns/providers/shared/request.go:1]: Why did you rename the package?
What is the difference between utils and shared? Is the package name sufficient and descriptive enough?

**Review by Valentin-Kaiser — CHANGES_REQUESTED** [review]

File: pkg/ddns/providers/shared/request.go (hunk 3 above)
Comment (Plaenkler) [reply pkg/ddns/providers/shared/request.go:1]: Any specific meaningful name would conflict with other modules. Like request, for example. Therefore I have assigned the code snippet to a package below the providers which share this code. Alternative suggestions are welcome.

Comment (Valentin-Kaiser) [reply pkg/ddns/providers/shared/request.go:1]: Why is it necessary to export this 25-line function, which is only used in the provider package, to another package?

**Review by Valentin-Kaiser — APPROVED** [review]
